### DISCUSSION TITLE
Call to undefined method App\Models\User::authorizedToViewAny()??

### DISCUSSION THREAD

**Discussion by octoxan** [laravel/nova-issues#4878] [Q&A]
Trying to get this thing installed, and keep running into very poor documentation.

Finally got it to think it's authorized on localhost, by setting the APP_URL to localhost. Would have been cool if the docs mentioned it was going off the .env and not what you're viewing it in the browser at.

Now, trying to get Resources to actually show up in the sidebar.

https://nova.laravel.com/docs/4.0/resources/

> By default, all resources within the app/Nova directory will automatically be registered with Nova. You are not required to manually register them.

This does not seem true? It comes with a User.php but there is no Resources sidebar in Nova. It says these can be manually registered by _overriding the resources method_ but there isn't even a resources method by default. Added one manually going off the docs, if they can be trusted:

```php
protected function resources()
    {
        Nova::resourcesIn(app_path('Nova'));

        Nova::resources([
            User::class,
        ]);
    }
```

Added this and now the site throws this error:

`Call to undefined method App\Models\User::authorizedToViewAny()`

**Comment by crynobone** [comment]
> Finally got it to think it's authorized on localhost, by setting the APP_URL to localhost. Would have been cool if the docs mentioned it was going off the .env and not what you're viewing it in the browser at.

Are you referring to Unregistered warning? not 100% sure here.

> Added one manually going off the docs, if they can be trusted:

The documentation clearly has `use App\Nova\User;` in the example and it seem you have `use App\Models\User;`

> This does not seem true? It comes with a User.php but there is no Resources sidebar in Nova.

Do you have any `UserPolicy` blocking access to `viewAny` policy to the User resource?

**Reply by octoxan** [reply]
Apparently I had to `art optimize:clear` and the error disappeared, and the resources showed up.

The error was coming from the Policy which does use `App\Models\User` and not `App\Nova\Users`.

I'm not sure why I had to run that artisan command on initial install for the resources to show up. It doesn't need ran on every change, just that initial set up I guess.
